src/settings.rs

```diff
@@ -44,88 +40,67 @@ impl Default for Settings {
-        s.set_default("limits.max_post_bytes", i64::from(DEFAULT_MAX_POST_BYTES))?;
-        s.set_default(
-            "limits.max_post_records",
-            i64::from(DEFAULT_MAX_POST_RECORDS),
-        )?;
-        s.set_default(
-            "limits.max_record_payload_bytes",
-            i64::from(DEFAULT_MAX_RECORD_PAYLOAD_BYTES),
-        )?;
-        s.set_default(
-            "limits.max_request_bytes",
-            i64::from(DEFAULT_MAX_REQUEST_BYTES),
-        )?;
-        s.set_default("limits.max_total_bytes", i64::from(DEFAULT_MAX_TOTAL_BYTES))?;
-        s.set_default(
-            "limits.max_total_records",
-            i64::from(DEFAULT_MAX_TOTAL_RECORDS),
-        )?;
-        s.set_default("statsd_host", "localhost")?;
-        s.set_default("statsd_port", 8125)?;
-        s.set_default("statsd_label", "tokenserver")?;
-        s.set_default(
-            "limits.default_oauth_scope",
-            String::from(DEFAULT_OAUTH_SCOPE),
-        )?;
+        // there appears to be a bug with `Config::try_from`
+        // https://github.com/mehcode/config-rs/issues/144
+        // let mut s = Config::try_from(&Settings::default())?;
+        let mut config = Config::new();
 
         // Merge the config file if supplied
         if let Some(config_filename) = filename {
-            s.merge(File::with_name(config_filename))?;
+            config.merge(File::with_name(config_filename))?;
         }
 
         // Merge the environment overrides
-        s.merge(Environment::with_prefix(PREFIX))?;
+        // Environment variables are usually upper case. Let's uppercase the
+        // PREFIX just to make sure.
+        config.merge(Environment::with_prefix(&PREFIX.to_ascii_uppercase()))?;
 
-        Ok(match s.try_into::<Self>() {
-            Ok(s) => {
-                // Adjust the max values if required.
-                s
-            }
-            Err(e) => match e {
-                // Configuration errors are not very sysop friendly, Try to make them
-                // a bit more 3AM useful.
-                ConfigError::Message(v) => {
-                    println!("Bad configuration: {:?}", &v);
-                    println!("Please set in config file or use environment variable.");
-                    println!(
-                        "For example to set `database_url` use env var `{}_DATABASE_URL`\n",
-                        PREFIX.to_uppercase()
-                    );
-                    error!("Configuration error: Value undefined {:?}", &v);
-                    return Err(ConfigError::NotFound(v));
-                }
-                _ => {
-                    error!("Configuration error: Other: {:?}", &e);
-                    return Err(e);
-                }
+        // hand merge the configuration and defaults into a consistent settings.
+        // bleh.
+        let default = Self::default();
+        Ok(Self {
+            debug: config.get_bool("config").unwrap_or(default.debug),
```

Comment:
FYI, turns out [there's another way](https://github.com/mehcode/config-rs/issues/144#issuecomment-662993649) to do this.

Reply:
Glad to see this. It makes code much cleaner.